File: java/입문/진료순서_정하기.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] emergency) {
        int[] sortEmergency = emergency.clone();
        Arrays.sort(sortEmergency);
        int[] answer = new int[emergency.length];
        for(int i = 0; i < answer.length; i++) {
            for(int j = sortEmergency.length - 1; j >= 0; j--) {
                if(sortEmergency[j] == emergency[i]) {
                    answer[i] = sortEmergency.length - j;
                    break;
                }
            }
        }
        return answer;
    }
}

File: swift/입문/진료순서_정하기.swift
// 진료순서 정하기 문제

import Foundation

func solution(_ emergency:[Int]) -> [Int] {
    let sort_emergency = emergency.sorted(by: >)
    return emergency.map { sort_emergency.firstIndex(of: $0)! + 1 }
}

// emergency.map { data in
//         emergency.filter { $0 > data }.count + 1
//     }